### PR TITLE
Add phase data reconstruction IC

### PR DESCRIPTION
**Pull request by dschwen**
This is the first part in addressing #5502. I created a new InitialCondition that initializes a variable to reflect the distribution of a _phase_ in an EBSD data set. I felt it would be better to split this out into another class, as the original ReconVarIC was starting to become a convoluted mess of boolean flags.

This adds a node to phase weight map, that is similar to the node to grain weight map. It helps determining what the weight of a phase at a node is (as EBSD data lives on the elements we need to average over elements adjacent to a node).

This PR re-enables the 2phase_reconstruction_test.i (I had to re-gold because the old gold file was crap. The new result is a smooth map).

Ping @tonkmr and @frombs
Refs #5502

### CODE REVIEW

File: modules/phase_field/src/ics/ReconPhaseVarIC.C [hunk 1]
```diff
@@ -0,0 +1,44 @@
+#include "ReconPhaseVarIC.h"
```
Comment (permcody) [inline modules/phase_field/src/ics/ReconPhaseVarIC.C]: Missing module header here

**Comment by dschwen** [comment]
Thanks @permcody, I added that. Civet failed the pre-test with 
```
Stopping at 'libmesh'; script returned non-zero status.
```
what up?!

**Comment by tonkmr** [comment]
I want to be clear on the use of the various ICs here. Correct me if I'm wrong on any of this.

If I want to assign all the grains from the EBSD file to order parameters and ignore phase, I use ReconVarICAction (and thus ReconVarIC) and don't provide a value for the phase input parameter.

If I want to assign only the grains from a specific phase from the EBSD file to order parameters , I use ReconVarICAction and provide the phase number (phase = 2).

If I want to assign all values for a specific phase to one variable and ignore the grain number, I use ReconPhaseVarIC.

Is that correct?

**Comment by dschwen** [comment]
Yes, @tonkmr, that is correct. Functionality number two is not addressed by this PR. That will be a followup PR. 1 and 3 are already working with this PR.

**Comment by dschwen** [comment]
I need someone to review this.

**Comment by permcody** [comment]
:+1:

**Pull request merged**